proto/viam/app/v1/app.proto

```diff
@@ -830,6 +832,57 @@ message CheckPermissionsResponse {
   repeated AuthorizedPermissions authorized_permissions = 1;
 }
 
+message ModuleVersion {
+  // The semver string that represents the major/minor/patch version of the module
+  string version = 1;
+  // The uploads that are available for this module version
+  repeated Uploads files = 2;
+  // The models that this verion of the module provides
+  repeated Model models = 3;
+  // The entrypoint for this version of the module
+  string entrypoint = 4;
+}
+
+message ModuleMetadata {
+  // A list of models that are available in the module
+  repeated Model models = 1;
+  // A list of versions of the module that are available
+  repeated ModuleVersion versions = 2;
```

Comment:
To solidify our offline discussion, if I am understanding this right, when you add ml models to the registry, you would add something like:
```
message MlModelMetadata {
   ....
    repeated MlModelVersionMetadata versions = xyz;
}
```
right?

(because I seem to recall the data team talking about how not all ml_model packages should turn into registry versions so you need metadata about which models actually ended up in the registry)

Reply:
This was my main thought on this PR. Any reason we can't add the ML Model versioning now? We know we need it and there won't be a way to show versions for ML Models without it (except for calling ListPackages which we don't want to do).

Reply:
no reason, just added!